lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -48,7 +48,8 @@ def matches?(given_proc, negative_expectation=false, &block)
           begin
             given_proc.call
           rescue Exception => @actual_error
-            if values_match?(@expected_error, @actual_error)
+            if values_match?(@expected_error, @actual_error) ||
+               (values_match?(@expected_error, @actual_error.message) rescue false)
```

Comment:
It's a bug in the custom matcher used in that spec. A matchers `matches?` should return `true` or `false` to indicate a `match`, and if it's nonsensical to match an object against it can return `false` from `matches?` and return a message explaining the nonsensicality in `failure_message`.

In this case, the `an_error_with_attribute` matcher should do a `error.respond_to?(attr)` check in it's `match` logic.

Does that make sense?